.github/workflows/run_mmperf.yml

```diff
@@ -0,0 +1,174 @@
+        with:
+          submodules: true
+      - name: "Building mmperf for CUDA"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+            gcr.io/iree-oss/mmperf@sha256:f1b458cb9570e5e70ffbd59d64d365ed818cb3b308d6e42a4cf5807f2c1a9ced \
+          ./build_tools/benchmarks/mmperf/build_mmperf.sh "${MMPERF_REPO_DIR}" "${BUILD_DIR}" "cuda"
+      - name: "Removing unused files"
+        run: |
+          find "${BUILD_DIR}" -type f -name "*.a" -o -type f -name "*.o" \
+            -print \
+            -delete
+      - name: "Creating build dir archive"
+        id: archive
+        env:
+          BUILD_DIR_ARCHIVE: ${{ env.BUILD_DIR }}.tar.zst
+        run: |
+          tar -I 'zstd -T0' \
+            -cf ${BUILD_DIR_ARCHIVE} ${BUILD_DIR}
+          echo "::set-output name=build-dir-archive::${BUILD_DIR_ARCHIVE}"
+      - name: "Uploading build dir archive"
+        id: upload
+        env:
+          BUILD_DIR_ARCHIVE: ${{ steps.archive.outputs.build-dir-archive }}
+          BUILD_DIR_GCS_ARTIFACT: ${{ env.GCS_DIR }}/${{ steps.archive.outputs.build-dir-archive }}
+        run: |
+          gcloud alpha storage cp "${BUILD_DIR_ARCHIVE}" "${BUILD_DIR_GCS_ARTIFACT}"
+          echo "::set-output name=build-dir-gcs-artifact::${BUILD_DIR_GCS_ARTIFACT}"
+
+  benchmark_cuda:
+    needs: [setup, build_cuda]
+    if: needs.setup.outputs.should-run == 'true'
+    runs-on:
+      - self-hosted  # must come first
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=${{ needs.setup.outputs.runner-env }}
+      - gpu
+      - os-family=Linux
+    env:
+      MMPERF_REPO_DIR: ${{ needs.setup.outputs.mmperf-repo-dir }}
+      RESULTS_DIR: mmperf-results-cuda
+      RESULTS_GCS_UPLOAD_DIR: "gs://mmperf-benchmark-artifacts/cuda"
+      BUILD_DIR: ${{ needs.build_cuda.outputs.build-dir }}
+      BUILD_DIR_ARCHIVE: ${{ needs.build_cuda.outputs.build-dir-archive }}
+      BUILD_DIR_GCS_ARTIFACT: ${{ needs.build_cuda.outputs.build-dir-gcs-artifact }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          submodules: true
+      - name: "Downloading build dir archive"
+        run: gcloud alpha storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
+      - name: "Extracting build dir archive"
+        run: tar -xf "${BUILD_DIR_ARCHIVE}"
+      - name: "Running mmperf on CUDA"
+        run: |
+          mkdir ${RESULTS_DIR}
+          ./build_tools/github_actions/docker_run.sh \
+          --gpus all \
+            gcr.io/iree-oss/mmperf@sha256:f1b458cb9570e5e70ffbd59d64d365ed818cb3b308d6e42a4cf5807f2c1a9ced \
+          ./build_tools/benchmarks/mmperf/run_mmperf.sh "${MMPERF_REPO_DIR}" "${BUILD_DIR}" "${RESULTS_DIR}"
+      - name: "Uploading results"
+        run: |
+          export GCS_DIR_NAME="$(date +'%Y-%m-%d').$(date +'%s')"
```

Comment:
We're computing this separately in the separate jobs. We don't want it to be the same? Should this be the commit hash instead? If you want to share it, you can set it as an output from the setup step

Reply:
Moved the calculation to the setup step and added the IREE sha.